metaflow/package.py

```diff
@@ -59,19 +60,60 @@ def _recurse(root):
 
 
 class MetaflowPackage(object):
-    def __init__(self, flow, environment, echo, suffixes=DEFAULT_SUFFIXES_LIST):
+    def __init__(
+        self,
+        flow,
+        environment,
+        echo,
+        suffixes=DEFAULT_SUFFIXES_LIST,
+        flow_datastore=None,
+        logger=None,
+    ):
         self.suffixes = list(set().union(suffixes, DEFAULT_SUFFIXES_LIST))
         self.environment = environment
         self.metaflow_root = os.path.dirname(__file__)
 
         self.flow_name = flow.name
         self._flow = flow
+        self.flow_datastore = flow_datastore
+        self.logger = logger
         self.create_time = time.time()
+        self._is_package_available = False
+        self.blob = None
+        self.package_url = None
+        self.package_sha = None
+
+        # Make package creation and upload asynchronous
+        self._init_thread = threading.Thread(
+            target=self._prepare_and_upload_package,
+            args=(flow, environment, flow_datastore, echo),
+        )
+        self._init_thread.daemon = True
+        self._init_thread.start()
+
+    def _prepare_and_upload_package(self, flow, environment, flow_datastore, echo):
```

Comment:
hardening: wrap in a try/except block to deal with issues like this: https://www.joeshaw.org/python-daemon-threads-considered-harmful/